spec/lib/pdf_filler/nj1040_pdf_spec.rb

```diff
@@ -317,202 +452,143 @@
-              primary_first_name: "Grace",
-              primary_last_name: "Hopper",
-              primary_middle_initial: "",
-              primary_suffix: "JR"
+              :spouse_veteran
             )
           }
-          it 'fills pdf with LastName FirstName Suf' do
-            expected_name = "Hopper Grace JR"
-            expect(pdf_fields[name_field]).to eq expected_name
-            expect(pdf_fields["Names as shown on Form NJ1040"]).to eq expected_name
-            expect(pdf_fields["Names as shown on Form NJ1040_2"]).to eq expected_name
-            expect(pdf_fields["Names as shown on Form NJ1040_3"]).to eq expected_name
+          it "checks the spouse veteran exemption but not the self checkbox" do
+            expect(pdf_fields["Check Box45"]).to eq "Off"
+            expect(pdf_fields["Check Box46"]).to eq "Yes"
+            expect(pdf_fields["x  6000"]).to eq "6000"
+            expect(pdf_fields["undefined_11"]).to eq "1"
           end
         end
 
-        context "with suffix and middle initial" do
+        context "primary and spouse are both veterans" do
           let(:submission) {
             create :efile_submission, tax_return: nil, data_source: create(
               :state_file_nj_intake,
-              primary_first_name: "Grace",
-              primary_last_name: "Hopper",
-              primary_middle_initial: "B",
-              primary_suffix: "JR"
+              :married_filing_jointly,
+              :primary_veteran,
+              :spouse_veteran
             )
           }
-          it 'fills pdf with LastName FirstName MI Suf' do
-            expected_name = "Hopper Grace B JR"
-            expect(pdf_fields[name_field]).to eq expected_name
-            expect(pdf_fields["Names as shown on Form NJ1040"]).to eq expected_name
-            expect(pdf_fields["Names as shown on Form NJ1040_2"]).to eq expected_name
-            expect(pdf_fields["Names as shown on Form NJ1040_3"]).to eq expected_name
+          it "claims both the self and spouse veteran exemptions" do
+            expect(pdf_fields["Check Box45"]).to eq "Yes"
+            expect(pdf_fields["Check Box46"]).to eq "Yes"
+            expect(pdf_fields["x  6000"]).to eq "12000"
+            expect(pdf_fields["undefined_11"]).to eq "2"
           end
         end
       end
 
-      context "joint filer" do
-        context "same last name" do
+      describe "Line 12 dependents attending college" do
+        context 'when has 2 dependents in college' do
           let(:submission) {
             create :efile_submission, tax_return: nil, data_source: create(
-              :state_file_nj_intake,
-              primary_first_name: "Bert",
-              primary_last_name: "Muppet",
-              primary_middle_initial: "S",
-              spouse_first_name: "Ernie",
-              spouse_last_name: "Muppet",
-              spouse_ssn: "123456789"
+              :state_file_nj_intake, :two_dependents_in_college
```

Comment:
I rearranged a lot of the tests in here because they were out of order with the PDF. No existing tests have been modified though, and this line 12 test is the only actual change that needs review. Everything else is just rearranging describe blocks